### PR TITLE
SUBMARINE-1187.  Refine the interface of internal service and keep more details into db

### PR DESCRIPTION
**Pull request by FatalLin**
### What is this PR for?
per discussion with @pingsutw, we would like to persist more detail data into database which allow us to provide to users when they check the detail, therefore I refine the logic when handler receiving the events, it will align the same logic in k8ssubmitter to rebuild the corresponding object, and pass to internal service, then the internal service persist it into database.

### What type of PR is it?
Improvement
### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-1187
### How should this be tested?
existed unit tests.
### Screenshots (if appropriate)
same of: https://github.com/apache/submarine/pull/877
### Questions:
* Do the license files need updating? No
* Are there breaking changes for older versions? No
* Does this need new documentation? No

### CODE REVIEW

File: submarine-server/server-core/pom.xml [hunk 1]
```diff
@@ -437,6 +437,11 @@
           <artifactId>jackson-annotations</artifactId>
         </exclusion>
       </exclusions>
+    </dependency>
+	<dependency>
```
Comment (jiwq) [inline submarine-server/server-core/pom.xml:441]: indentation nit~

File: submarine-server/server-core/pom.xml [hunk 2]
```diff
@@ -437,6 +437,11 @@
...
+      <groupId>joda-time</groupId>
+      <artifactId>joda-time</artifactId>
+      <version>2.10.8</version>
```
Comment (jiwq) [inline submarine-server/server-core/pom.xml:444]: - should better move the dependency to center management
- the artifact's version should define in property

File: submarine-server/server-core/src/main/java/org/apache/submarine/server/experiment/database/entity/ExperimentEntity.java [hunk 3]
```diff
@@ -19,6 +19,7 @@
 
 package org.apache.submarine.server.experiment.database.entity;
 
+
```
Comment (jiwq) [inline submarine-server/server-core/src/main/java/org/apache/submarine/server/experiment/database/entity/ExperimentEntity.java:22]: redundant blank line

**Review by pingsutw — APPROVED** [review]
+1, Thank you @FatalLin and @jiwq

**Pull request closed without merging**